datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```
@@ -944,6 +957,101 @@ private void handleLeaderDoAssignment() {
+      Map<String, Set<DatastreamTask>> assignmentByInstance = new HashMap<>(previousAssignmentByInstance);
+
+      // retrieve the datastreamGroups for validatio
+      DatastreamGroup toProcessDatastream =
+          fetchDatastreamGroups().stream().filter(dg -> datastreamGroupName.equals(dg.getName())).findFirst().orElse(null);
+
+      if (toProcessDatastream != null) {
+        AssignmentStrategy strategy = _connectors.get(toProcessDatastream.getConnectorName()).getAssignmentStrategy();
+        Connector connectorInstance = _connectors.get(toProcessDatastream.getConnectorName()).getConnector()
+            .getConnectorInstance();
+        Map<String, Optional<DatastreamGroupPartitionsMetadata>> datastreamPartitions =
+            connectorInstance.getDatastreamPartitions();
+
+        if (datastreamPartitions.containsKey(toProcessDatastream.getName())) {
+          DatastreamGroupPartitionsMetadata subscribes = connectorInstance.getDatastreamPartitions()
+              .get(toProcessDatastream.getName())
+              .orElseThrow(() ->
+                  new DatastreamTransientException("Subscribed partition is not ready yet for datastream " +
+                      toProcessDatastream.getName()));
+
+          assignmentByInstance = strategy.assignPartitions(assignmentByInstance, subscribes);
+        } else {
+          _log.warn("partitions for {} is not found, ignore the partition assignment", toProcessDatastream.getName());
+        }
+      }
+
+      for (String key : assignmentByInstance.keySet()) {
+        newAssignmentsByInstance.put(key, new ArrayList<>(assignmentByInstance.get(key)));
+      }
+      _adapter.updateAllAssignments(newAssignmentsByInstance);
+      _log.info("Partition assignment completed: datastreamGroup, assignment {} ", assignmentByInstance);
+      succeeded = true;
+    } catch (Exception ex) {
+      _log.info("Partition assignment failed, Exception: ", ex);
```

Why are we swallowing the exception here? This will result in new partitions not being consumed right?

---

If there is any exception, the succeeded flag will still be false, and the retry logic will be executed. I can make it clear by assigning the succeeded to false in the exception handling part.